### PR TITLE
bt-169: Create talent sign up third step form

### LINKED ISSUE #169: Create Talent Sign up third step form
- [x] Add to bundle components step3 form
- [x] On step3 form Next button data should be validated, use useAppForm() hook for this
- [ ] After validation save data in Store

Image

**User story:**
As a talent, I want to be able to fill up a profile with info about my skills, so that potential employers can evaluate my candidacy.

**Acceptance criteria:**
- Text “STEP 03/05” is placed in the upper right corner on the same line as the Skills and projects.
- The third step must include input fields:
1. Hard Skills (A-Z, a-z, special chars, spaces)
2. Level of English dropdown	(No English, A1 Elementary, A2 Pre-intermediate, B1 Intermediate, B2 Upper intermediate, C1 Advanced, C2 Proficiency)
3. I do not consider	checkboxes (Gambling, Dating, Gamedev, Crypto, Startups)
4. Preferred language of communication dropdown (English, Ukrainian, Russian, Spanish, Portuguese, French, German, and Chinese (Mandarin))
5. Project link URL (5-50 symbols, A-Z, a-z, 0-9, special chars). User can add up to 5 links.
- The Hard Skills, Level of English, I do not consider, Preferred language of communication are required and have “*”
- Text inputs and dropdowns should have a placeholder
- Any other symbols in text fields or unsupported length, format should trigger the validation message appearance
- Talent should be able to back to the previous step BSA badges by clicking on the Back button
- Talent should be able to go to the next step CV and contacts by clicking on the Next button. Only if no validation errors are present.
- Talent's information should be saved in the fields when switching between steps

### PR DESCRIPTION
**Pull request by lavandosya**
Screenshot 2023-09-06 at 18 28 47
Screenshot 2023-09-06 at 18 29 02

### CODE REVIEW

**Comment by nikita-remeslov** [comment]
@lavandosya link PR in the ticket

**Comment by nikita-remeslov** [comment]
@lavandosya merge development branch pls

File: frontend/src/bundles/common/components/input/input.tsx [hunk 1]
```diff
@@ -25,6 +26,7 @@ type Properties<T extends FieldValues> = {
     adornmentText?: string;
     className?: string;
     inputClassNames?: string;
+    inputRef?: RefCallBack;
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/input/input.tsx:29]:
```suggestion
    ref?: RefCallBack;
```

File: frontend/src/bundles/talent-onboarding/components/skills-step/constants/default.constants.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { type SkillsStepDto } from '../../../../../../../shared/src/bundles/talent-onboarding/types/skills-step/skills-step-dto.js';
```
Comment (nikita-remeslov) [inline frontend/src/bundles/talent-onboarding/components/skills-step/constants/default.constants.ts]: you should use aliad. DTO - Should the importer form `build` folder

check - frontend/src/bundles/users/types/types.ts

File: frontend/src/bundles/talent-onboarding/store/slice.ts [hunk 3]
```diff
@@ -1,31 +1,43 @@
...
 const { reducer, actions, name } = createSlice({
     initialState,
-    name: 'signUp',
+    name: 'talentOnBoarding',
     reducers: {},
     extraReducers(builder) {
         builder.addCase(profileStep.fulfilled, (state, action) => {
+            state.profileStep = action.payload;
+        });
+        builder.addCase(skillsStep.fulfilled, (state, action) => {
```
Comment (nikita-remeslov) [inline frontend/src/bundles/talent-onboarding/store/slice.ts:34]: not sure about this logic.

Comment (nikita-remeslov) [reply frontend/src/bundles/talent-onboarding/store/slice.ts:34]: I think we can same all the steps in on entity

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/input/input.tsx (hunk 1 above)
Comment (lavandosya) [reply frontend/src/bundles/common/components/input/input.tsx:29]: this should be named exactly like inputRef, otherwise errors will occur

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**